Sources/Events/EventsManager.swift

```diff
@@ -114,6 +115,39 @@ actor EventsManager: EventsManagerType {
     }
     #endif
 
+    func flushEventsWithBackgroundTask() async {
+        #if os(iOS) || os(tvOS) || VISION_OS
+        let endBackgroundTask = await self.beginBackgroundTask(named: "com.revenuecat.flushEvents")
+        defer {
+            endBackgroundTask?()
+        }
+        #endif
+
+        do {
+            _ = try await self.flushEvents(batchSize: Self.defaultEventBatchSize)
+        } catch {
+            Logger.error(Strings.paywalls.event_flush_failed(error))
+        }
+    }
+
+    func flushFeatureEventsWithBackgroundTask() async {
+        #if os(iOS) || os(tvOS) || VISION_OS
+        let endBackgroundTask = await self.beginBackgroundTask(named: "com.revenuecat.flushFeatureEvents")
+        defer {
+            endBackgroundTask?()
+        }
+        #endif
+
+        do {
+            _ = try await self.flushFeatureEvents(batchSize: Self.defaultEventBatchSize)
+        } catch {
+            Logger.error(Strings.paywalls.event_flush_failed(error))
+        }
+    }
+
+    // MARK: - Internal Methods (for testing)
+    // External consumers should use flushEventsWithBackgroundTask() and flushFeatureEventsWithBackgroundTask()
```

Comment:
This does look like a smell... ideally we should be only testing public methods... But I think it's ok for now since we want to release this asap.